src/main/java/farmsystem/backend/domain/member/service/MemberService.java

```diff
@@ -0,0 +1,65 @@
+package farmsystem.backend.domain.member.service;
+
+import farmsystem.backend.domain.member.dto.request.MemberLoginRequest;
+import farmsystem.backend.domain.member.dto.request.MemberSignupRequest;
+import farmsystem.backend.domain.member.dto.response.MemberLoginResponse;
+import farmsystem.backend.domain.member.dto.response.MemberResponse;
+import farmsystem.backend.domain.member.entity.Member;
+import farmsystem.backend.domain.member.repository.MemberRepository;
+import farmsystem.backend.domain.profile.entity.Profile;
+import farmsystem.backend.domain.profile.entity.ProfileType;
+import farmsystem.backend.global.common.JwtUtil;
+import farmsystem.backend.global.error.ErrorCode;
+import farmsystem.backend.global.error.exception.ConflictException;
+import farmsystem.backend.global.error.exception.EntityNotFoundException;
+import farmsystem.backend.global.error.exception.InvalidValueException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDate;
+
+@Service
+@RequiredArgsConstructor
+public class MemberService {
+    private final PasswordEncoder passwordEncoder;
+    private final MemberRepository memberRepository;
+    private final JwtUtil jwtUtil;
+
+    @Transactional
+    public MemberResponse signUp(MemberSignupRequest memberSignupRequest) {
+        if(memberRepository.existsByEmail(memberSignupRequest.email())){
+            throw new ConflictException(ErrorCode.CONFLICT);
+        }
+
+        String encodedPassword = passwordEncoder.encode(memberSignupRequest.password());
+
+        Member memberToSave = memberSignupRequest.toEntity((encodedPassword));
+        Profile profile = Profile.builder()
+                .member(memberToSave)
+                .name(memberToSave.getUsername())
+                .type(ProfileType.LIVE)
+                .balance(1000000)
+                .startDate(LocalDate.now())
+                .build();
+        memberToSave.addProfile(profile);
+
+        Member member = memberRepository.save(memberToSave);
+        return MemberResponse.from(member);
+    }
+
+    @Transactional
```

Comment:
login에서는 읽기전용으로 하면 될 것 같아요~
```suggestion
    @Transactional(readOnly=true)
```

Reply:
아니면 클래스 위에 저렇게 선언해서 클래스를 기본적으로 readOnly가 되게 하고(따로 선언 안한 메서드는 readOnly) readOnly 아닌 애들만 @Transactional 붙여서 덮어 씌울 수 있어요

Reply:
아하 감사합니다!